.github/workflows/codeql.yml

```diff
@@ -0,0 +1,46 @@
+name: "Code Scanning - Action"
+
+on:
+  push:
+  schedule:
+    - cron: '0 0 * * 0'
+
+jobs:
+  CodeQL-Build:
+
+    strategy:
+      fail-fast: false
+
+
+    # CodeQL runs on ubuntu-latest, windows-latest, and macos-latest
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      # Override language selection by uncommenting this and choosing your languages
+      # with:
+      #   languages: go, javascript, csharp, python, cpp, java
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below).
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
+
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v1
```

Comment:
Will the `analyze` step need to use the compiled binary from the previous step? If so, how will it know the location of the binary?

Reply:
Nope, analyze works in one of two ways: 

1. For scanned languages (Go, JavaScript, TypeScript, Python), it scans the code in the repository without building it
2. For traced languages, a database which represents your code is created during compilation. This is because the init step hooks into the build tasks to monitor the builds as they take place.

Reply:
@jhutchings1 Can you kindly refer me to the part of the documentation that tells which languages are analyzed using scanning and which not (i.e. require build task)? In the future, when more languages get added, this will help me cross-reference and find out whether to enable `autobuild` step or not.